variables.tf

```diff
@@ -32,94 +44,105 @@ variable "allowed_ssh_ranges" {
-  default     = true
-}
+variable "kms" {
+  description = <<EOF
+  kms = {
+    database_encryption_key_name = Use an existing KMS key for the Application-layer Secrets Encryption settings. (Optional)
+  }
+  EOF
 
-variable "enable_network_policy" {
-  description = "Enable network policy switch. Has no effect when enable_advanced_datapath is true"
-  type        = bool
-  default     = true
-}
+  type = object({
+    database_encryption_key_name = optional(string, null)
+  })
 
-variable "enable_vertical_pod_autoscaling" {
-  type        = bool
-  default     = true
-  description = "Enable GKE vertical scaling"
+  default = {}
 }
 
-variable "gke_release_channel" {
-  type        = string
-  default     = "STABLE"
-  description = "GKE K8s release channel for master"
-}
+variable "gke" {
+  description = <<EOF
+  gke = {
+    k8s_version = Cluster k8s version
+    release_channel = GKE release channel
+    public_access = {
+      enabled = Enable API public endpoint
+      cidrs = List of CIDR ranges permitted for accessing the public endpoint
+    }
+    control_plane_ports =  Firewall ports to open from the master, e.g., webhooks
+    advanced_datapath = Enable the ADVANCED_DATAPATH provider
+    network_policies = Enable network policy switch. Has no effect when enable_advanced_datapath is true
+    vertical_pod_autoscaling = Enable GKE vertical scaling
+    kubeconfig = {
+      path = Specify where the cluster kubeconfig file should be generated.
+    }
+  }
+  EOF
 
-variable "location" {
-  type        = string
-  default     = "us-west1-b"
-  description = "The location (region or zone) of the cluster. A zone creates a single master. Specifying a region creates replicated masters accross all zones"
-}
+  type = object({
+    k8s_version     = optional(string, ""),
+    release_channel = optional(string, "RAPID"),
+    public_access = optional(object({
+      enabled = optional(bool, false),
+      cidrs   = optional(list(string), [])
+    }), {}),
+    control_plane_ports      = optional(list(string), [])
+    advanced_datapath        = optional(bool, true),
+    network_policies         = optional(bool, false),
+    vertical_pod_autoscaling = optional(bool, true),
+    kubeconfig = optional(object({
+      path = optional(string, null)
+    }), {})
+  })
 
-variable "master_firewall_ports" {
-  type        = list(string)
-  default     = []
-  description = "Firewall ports to open from the master, e.g., webhooks"
-}
+  default = {}
 
-variable "master_authorized_networks_config" {
-  type = list(object({
-    cidr_block   = string
-    display_name = string
-  }))
-  default = [
-    {
-      cidr_block   = "0.0.0.0/0"
-      display_name = "global-access"
-    }
-  ]
-  description = "Configuration options for master authorized networks. Default is for debugging only, and should be removed for production."
+  validation {
+    condition     = !var.gke.advanced_datapath || (var.gke.advanced_datapath && !var.gke.network_policies)
```

Comment:
this doesn't match your comment
`Enable network policy switch. Has no effect when enable_advanced_datapath is true`